test/common/upstream/outlier_detection_impl_test.cc

```diff
@@ -746,65 +777,70 @@ TEST(OutlierDetectionEventLoggerImplTest, All) {
   std::shared_ptr<MockHostDescription> host(new NiceMock<MockHostDescription>());
   ON_CALL(*host, cluster()).WillByDefault(ReturnRef(cluster));
   Event::SimulatedTimeSystem time_system;
-  time_system.setSystemTime(SystemTime()); // Matches '1970' absolute golden times below.
+  // This is rendered as "2018-12-18T09:00:00Z".a
+  time_system.setSystemTime(std::chrono::milliseconds(1545123600000));
   absl::optional<MonotonicTime> monotonic_time;
   NiceMock<MockDetector> detector;
 
   EXPECT_CALL(log_manager, createAccessLog("foo")).WillOnce(Return(file));
-  EventLoggerImpl event_logger(log_manager, "foo", time_system);
+  OutlierDetectionEventLoggerImpl event_logger(log_manager, "foo", time_system);
 
   StringViewSaver log1;
   EXPECT_CALL(host->outlier_detector_, lastUnejectionTime()).WillOnce(ReturnRef(monotonic_time));
-  EXPECT_CALL(*file, write(absl::string_view(
-                         "{\"time\": \"1970-01-01T00:00:00.000Z\", \"secs_since_last_action\": "
-                         "\"-1\", \"cluster\": "
-                         "\"fake_cluster\", \"upstream_url\": \"10.0.0.1:443\", \"action\": "
-                         "\"eject\", \"type\": \"5xx\", \"num_ejections\": 0, "
-                         "\"enforced\": \"true\"}\n")))
+
+  EXPECT_CALL(*file,
+              write(absl::string_view("{\"type\":\"CONSECUTIVE_5XX\",\"secs_since_last_action\":"
```

Comment:
a thought: I wonder if the change in output for the type string should be considered a deprecation? We might have consumers that string match in their observability tooling. wdyt @jmarantz?

Reply:
How do these deprecations typically work in Envoy? It seems like it might be hard to have any PR which just deletes the old mechanisms and switches over to the new mechanism; that you might need a period of time where both are supported via flag-flip, before you can finally remove the old mechanism after (say) the next point release.

My understanding is that in Envoy we don't typically do these flag-flip migrations but I'm wondering what the alternative is?